### DISCUSSION TITLE
关于图床畸形的成本划分以及个人变故

### DISCUSSION THREAD

**Discussion by z0z0r4** [LTCatt/PCL2Help#342] [General]
就是说，我现在马上高三了，根本没时间做啥项目，服务器空转，只有个图床
然而图片下载源依旧走的CDN，所以30r/m的服务器只为公开上传的入口而提供，付了两份钱，很亏，没意义，直接从香港服务器提供图片源又可能效果不咋地，可一年的CDN费用大概才 3G/d * 365 ~= 1T -> 125r，图床这个公开入口的价值比图片下载源本身还贵

寒假可能会开个大活攒pixiv图片特征，攒够了做图片反向搜索，这样机子一直爬虫肯定会满负荷，就会续到大一

如果这个想法放弃的话估计就直接抛了，图床入口估计也会关，但图有可能会继续提供，毕竟图可以放对象存储？看心情.jpg

起码寒假结束前我不会动，提前知会一声

~~可能性概况：无碍 -> 关图床维持图片 -> see you again~~

**Comment by z0z0r4** [comment]
如果各位撰写者内部有联系群我很乐意参考下各位的看法.jpg

~~矛盾点在于公开上传入口必须得有个服务器挂着，我又肯定不能把对象存储做成公共写，所以如果有这样的内部交流我很乐意直接开个 RAM 给常驻的哥们/leader来手动上传，守着个一年 34r*12 的传家宝服务器太蠢了，这么点图~~

**Reply by z0z0r4** [reply]
CDN 图片源完全可以维持下去，就是最懒的话我会把上传关了，没影响

**Comment by WorldHim** [comment]
所以 github 套 cdn 来做图床没有可行性吗🤔

**Reply by z0z0r4** [reply]
欸，完全有，正解，但同样得有个谁来拿权限就是了

**Reply by z0z0r4** [reply]
这样的话我还能顺便自动压缩

**Reply by z0z0r4** [reply]
ok，我来开个repo

**Reply by z0z0r4** [reply]
https://github.com/z0z0r4/PCL2HELPIMG

**Reply by WorldHim** [reply]
> 欸，完全有，正解，但同样得有个谁来拿权限就是了

组个团队？
指 GitHub Team

**Reply by z0z0r4** [reply]
团队...也好像没法申请加入的吧？我不了解

**Reply by z0z0r4** [reply]
GitHub team...想不出理由，感觉不如直接给PR权限

**Reply by WorldHim** [reply]
> 团队...也好像没法申请加入的吧？我不了解

应该可以把仓库转到团队然后给予成员权限吧🤔
这样权限管理也方便，而且看起来更正式？

**Reply by z0z0r4** [reply]
然而这会显得很怪异...只为了一个上传到图床而产生的 team，开个 PCL2HELP TEAM 把这个仓库也塞进去还差不多...

@LTCatt

**Reply by LTCatt** [reply]
umm 做个 bot 接 github api 自动过 pull request，或者自动上传到 repo 也行？

**Reply by z0z0r4** [reply]
有道理，可以我写个 workflow 自动替换 xml 里面的图片链接，secret 就存这仓库了，这样啥都不用

如何？调仓库 action @LTCatt 如果可以我下周末整个

**Reply by LTCatt** [reply]
我的意思是，用 CDN 代理 GitHub，然后用 bot 自动上传图片到 GitHub Repo，这样你就不需要服务器了（

**Reply by z0z0r4** [reply]
啥是用bot...感觉好麻烦，我就按上面的可？

**Reply by LTCatt** [reply]
也行

**Reply by WorldHim** [reply]
> 啥是用bot...感觉好麻烦，我就按上面的可？

所以应该怎么传图片🤔

**Reply by z0z0r4** [reply]
你直接随便找个图床上传，我 workflow 负责上传完替换 commit

**Reply by WorldHim** [reply]
> 你直接随便找个图床上传，我 workflow 负责上传完替换 commit

那为什么不直接传 repo，让 cdn 镜像 repo 内容👀

**Reply by z0z0r4** [reply]
我不想开 repo 负责这种东西，你们又没人负责管理仓库，我只打算cdn，lm 又懒得开官方

**Reply by LTCatt** [reply]
我的意思是，你可以开一个 repo，然后大家往 repo 交 issue，机器人就自动把 issue 里的图上传到 repo 里

**Reply by z0z0r4** [reply]
我不想自己开repo，不是很想负责这个...

何况 GitHub bot我就也不会用，还得有机子挂webhook，还得鉴权...

**Reply by z0z0r4** [reply]
不说了，等我寒假提交个 workflow 就这样吧...

**Comment by z0z0r4** [comment]
等龙猫看法吧...

我不是很打算自己维护这repo，期待lm单开个org把仓库和贡献者都加进去，我只提供加速，你们来管理 =+=

大概思路在这 https://github.com/z0z0r4/PCL2HELPIMG

~~没人管理我就懒得理了，这还挺麻烦的，干脆直接关了上传~~

**Reply by z0z0r4** [reply]
那这个就算了，麻烦
